File: Dot examples/ASP Book Examples/Book Examples/Chapter33/cs/App_Code/Glow.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Collections.Generic;

namespace MyControls
{
public class Glow : ScriptControl
{
    private string _Text = "Glow Control";
    private string _BackgroundColor = "orange";

    public string Text
    {
        get { return _Text; }
        set { _Text = value; }
    }

    public string BackgroundColor
    {
        get { return _BackgroundColor; }
        set { _BackgroundColor = value; }
    }

    protected override IEnumerable<ScriptReference> GetScriptReferences()
    {
        ScriptReference sref = new ScriptReference("Glow.js");
        List<ScriptReference> colSRefs = new List<ScriptReference>();
        colSRefs.Add(sref);
        return colSRefs;
    }

    protected override IEnumerable<ScriptDescriptor> GetScriptDescriptors()
    {
        ScriptControlDescriptor des = new ScriptControlDescriptor("myControls.Glow", this.ClientID);
        des.AddProperty("text", _Text);
        des.AddProperty("backgroundColor", _BackgroundColor);
        List<ScriptDescriptor> colDes = new List<ScriptDescriptor>();
        colDes.Add(des);
        return colDes;        
    }
}
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter33/vb/App_Code/Glow.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Collections.Generic

Namespace MyControls

Public Class Glow
    Inherits ScriptControl

    Private _Text As String = "Glow Control"
    Private _BackgroundColor As String = "orange"

    Public Property Text() As String
        Get
            Return _Text
        End Get
        Set(ByVal value As String)
            _Text = value
        End Set
    End Property

    Public Property BackgroundColor() As String
        Get
            Return _BackgroundColor
        End Get
        Set(ByVal value As String)
            _BackgroundColor = value
        End Set
    End Property

    Protected Overrides Function GetScriptReferences() As IEnumerable(Of ScriptReference)
        Dim sref As New ScriptReference("Glow.js")
        Dim colSRefs As New List(Of ScriptReference)()
        colSRefs.Add(sref)
        Return colSRefs
    End Function

    Protected Overrides Function GetScriptDescriptors() As IEnumerable(Of ScriptDescriptor)
        Dim des As New ScriptControlDescriptor("myControls.Glow", Me.ClientID)
        des.AddProperty("text", _Text)
        des.AddProperty("backgroundColor", _BackgroundColor)
        Dim colDes As New List(Of ScriptDescriptor)()
        colDes.Add(des)
        Return colDes
    End Function
End Class
End Namespace